### ISSUE TITLE
containers are not properly displayed when they are members of a user-defined class

### ISSUE THREAD

**Issue by yohm** [renatoGarcia/icecream-cpp#14]
I'm trying to print an instance of a user-defined class as follows.

```cpp
  class MyStruct {
    public:
    int i;
    double d;
    std::string s;
    std::vector<int> v;
  };

  MyStruct x;
  x.i = 2; x.d = 0.9; x.s = std::string("abc"); x.v = {1, 2, 3};     // set member variables
  IC(x);
```

What I expected was something like

```
ic| x: {i: 2, d: 0.9, s: "abc", v: [1, 2, 3]}
```

but what I actually got was the following.

```
ic| x: {i: 2, d: 0.9, s: {__r_: {}}, v: {}}
```

It seems that a member variable is not properly displayed when it is an STL container. If I print each member `IC(x.s, x.v)`, I got the correct output. Could someone investigate this issue?

FYI, my compiler is Apple clang version 11.0.0 (clang-1100.0.33.17).

**Comment by renatoGarcia** [comment]
Unfortunately this is the expected behavior.

When using the clang's `__builtin_dump_struct` function we have no access to each attribute, so we can print each one independently. That function supports only C style structures (no inheritance with split attributes between base and derived is supported, for example), and it will supply the info to be printed using the `printf` format specifiers.

In short, all we have is what clang give to us, and that is it. So the better we have is the implementation details of std::string and std::vector, if in that specific STL implementation that classes are StandardLayoutType. Otherwise the code won't even compile.

**Comment by yohm** [comment]
I got it. I'll accept it as a limitation. Thanks for your answer.
